src/components/ui/ChannelListModal.tsx

```diff
@@ -134,16 +174,81 @@ const ChannelListModal: React.FC = () => {
 
   useEffect(() => {
     if (selectedServerId) {
-      setIsLoading(true);
       listChannels(selectedServerId);
     }
   }, [selectedServerId, listChannels]);
 
+  // Sync filter state with store
+  useEffect(() => {
+    if (selectedServerId && channelListFilters[selectedServerId]) {
+      const filters = channelListFilters[selectedServerId];
+      setMinUsers(filters.minUsers || 0);
+      setMaxUsers(filters.maxUsers || 0);
+      setMinCreationTime(filters.minCreationTime || 0);
+      setMaxCreationTime(filters.maxCreationTime || 0);
+      setMinTopicTime(filters.minTopicTime || 0);
+      setMaxTopicTime(filters.maxTopicTime || 0);
+      setMask(filters.mask || "");
+      setNotMask(filters.notMask || "");
+    }
+  }, [selectedServerId, channelListFilters]);
+
+  // Scroll detection for lazy loading
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    const scrollContainer = scrollContainerRef.current;
+    if (!scrollContainer) return;
+
+    const handleScroll = () => {
+      const { scrollTop, scrollHeight, clientHeight } = scrollContainer;
+      const isNearBottom = scrollTop + clientHeight >= scrollHeight - 100; // 100px threshold
+
+      if (
+        isNearBottom &&
+        !loadingMoreRef.current &&
+        displayedCountRef.current < filteredChannels.length
+      ) {
+        setLoadingMore(true);
+        // Load next 50 channels
+        setTimeout(() => {
+          setDisplayedChannelsCount((prev) =>
+            Math.min(prev + 50, filteredChannels.length),
+          );
+          setLoadingMore(false);
+        }, 200); // Small delay for smooth UX
+      }
+    };
+
+    scrollContainer.addEventListener("scroll", handleScroll);
+    return () => scrollContainer.removeEventListener("scroll", handleScroll);
+  }, [filteredChannels.length, setDisplayedChannelsCount, setLoadingMore]); // Only depend on filteredChannels.length to avoid recreating listener too often
+
+  // Reset displayed count when filtered channels change
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think you can update react and use `useEffectEvent` for this now. https://react.dev/reference/react/useEffectEvent